bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/internal/YamlModelRepositoryImpl.java

```diff
@@ -153,13 +154,19 @@ public void deactivate() {
         watchService.unregisterListener(this);
     }
 
+    // Method for unit tests
+    protected void ignorePathRestrictions() {
+        pathRestrictionsIgnored = true;
+    }
+
     // The method is "synchronized" to avoid concurrent files processing
     @Override
     @SuppressWarnings({ "rawtypes", "unchecked" })
     public synchronized void processWatchEvent(Kind kind, Path path) {
         Path fullPath = watchPath.resolve(path);
         String modelName = path.toString();
-        if (path.startsWith("automation") || !modelName.endsWith(".yaml")) {
+        if (!pathRestrictionsIgnored && (!modelName.endsWith(".yaml") || !(path.startsWith("tags")
+                || path.startsWith("items") || path.startsWith("things") || path.startsWith("yaml")))) {
```

Comment:
Why not define an array constant with all the allowed directories at the top of the file? It may be easier to maintain when  new functionality is added.

Reply:
Would be better, I agree.